File: practice/algorithms/implementation/save_the_prisoner/SaveThePrisoner.scala
object SaveThePrisoner extends App {
  val lines = Source.stdin.getLines().toList
  val statements = lines.tail
  val savedPrisoners = statements.map(savedPrisoner)

  println(savedPrisoners.mkString("\n"))

  def savedPrisoner(line: String): Long = {
    val numbers = line.split(" ").map(_.toLong)
    val prisoners = numbers(0)
    val sweets = numbers(1)
    val startId = numbers(2)

    val saved = (((sweets % prisoners) + startId) % prisoners) - 1
    if (saved == 0) prisoners else saved
  }
}

File: practice/algorithms/implementation/save_the_prisoner/save_the_prisoner.rb
def saved_prisoner(line)
  numbers = line.split(" ").map {|i| i.to_i}
  prisoners = numbers[0]
  sweets = numbers[1]
  startId = numbers[2]

  saved = (((sweets % prisoners) + startId) % prisoners) - 1;
  return prisoners if (saved == 0)
  saved
end

n = gets.to_i
n.times do
  puts saved_prisoner(gets)
end